src/html/move-style-into-template.ts

```diff
@@ -45,24 +61,164 @@ class MoveStyleIntoTemplate extends HtmlRule {
           <dom-module>
   `);
 
-  async checkDocument(parsedDocument: ParsedHtmlDocument) {
-    const warnings: Warning[] = [];
-    const outOfPlaceStyle = p.AND(
-        p.hasTagName('style'), p.parentMatches(p.hasTagName('dom-module')));
-    const outOfPlaceStyles =
-        dom5.nodeWalkAll(parsedDocument.ast, outOfPlaceStyle);
-    for (const outOfPlaceNode of outOfPlaceStyles) {
-      warnings.push(new Warning({
-        parsedDocument,
-        code: this.code,
-        message:
-            `<style> tags should not be direct children of <dom-module>, they should be moved into the <template>`,
-        severity: Severity.WARNING,
-        sourceRange: parsedDocument.sourceRangeForNode(outOfPlaceNode)!
-      }));
+  async checkDocument(parsedDocument: ParsedHtmlDocument, document: Document) {
+    const warnings: FixableWarning[] = [];
+    const domModules = document.getFeatures({kind: 'dom-module'});
+    for (const domModule of domModules) {
+      const template = (domModule.astNode.childNodes ||
+                        []).find((n) => n.tagName === 'template');
+      if (!template) {
+        continue;
+      }
+      const moduleChildren = domModule.astNode.childNodes || [];
+      for (const child of moduleChildren) {
+        if (!styleMustBeInside(child)) {
+          continue;
+        }
+        const warning = new FixableWarning({
+          parsedDocument,
+          code: this.code,
+          message:
+              `Style tags should not be direct children of <dom-module>, they should be moved into the <template>`,
+          severity: Severity.WARNING,
+          sourceRange: parsedDocument.sourceRangeForStartTag(child)!
+        });
+        warnings.push(warning);
+
+        const templateContentStart =
+            parsedDocument.sourceRangeForStartTag(template)!.end;
+
+        const styleIndentation = getIndentationInside(child);
+        const templateIndentation = getIndentationInside(
+            parse5.treeAdapters.default.getTemplateContent(template));
+        const clonedStyle = clone(child);
+        const contents = clonedStyle.childNodes;
+        if (styleIndentation === templateIndentation && contents != null &&
+            contents.every(dom5.isTextNode)) {
```

Comment:
why is this `contents.every(dom5.isTextNode)` check necessary?

Reply:
Maybe it's paranoid, but it's to be sure that there aren't any elements inside the `<style>` node.

I mean, there shouldn't ever be, but... just in case!

I moved the check into the new `addIndentation` function